katcp/test/test_core.py

```diff
@@ -490,3 +490,42 @@ def call_set():
         # Test for NOT timing out
         self.io_loop.add_callback(call_set)
         yield self._event.until_set(timeout=0.1)
+
+
+class TestUntilSome(tornado.testing.AsyncTestCase):
+
+    @tornado.testing.gen_test
+    def test_until_some_args(self):
+        results = yield until_some()
+        self.assertEqual(results, [], 'Expected empty list for until_some()')
+        f1 = tornado.concurrent.Future()
+        f2 = tornado.concurrent.Future()
+        f3 = tornado.concurrent.Future()
+        # Test for timing out
+        with self.assertRaises(tornado.gen.TimeoutError):
+            yield until_some(f1, f2, f3, timeout=0.05)
+        # Test for NOT timing out
+        f3.set_result(84)
+        f1.set_result(24)
+        f2.set_result(42)
+        results = yield until_some(f1, f2, f3, timeout=0.1)
+        self.assertEqual(results, [(0, 24), (1, 42), (2, 84)],
```

Comment:
Smoothed the rough edges in a pending commit. Put generous warnings that 1.0 != 1 .

The futures are coming out in index order if they were resolved before WaitIterator starts, but it is an implementation feature, so I'll modify accordingly.